test/DynamoCoreTests/UserDataMigrationTests.cs

```diff
@@ -166,5 +170,118 @@ public void GetSortedInstalledVersions_FromDirectoryInspection()
             }
             
         }
+
+        private static void CreateMockPreferenceSettingsFile(string filePath, string packageDir)
+        {
+            var settings = new PreferenceSettings
+            {
+                CustomPackageFolders = new List<string>{packageDir}
+            };
+            settings.Save(filePath);
+        }
+
+        /// <summary>
+        /// Create 0.8, and 0.9 version directories in Temp dir
+        /// Create dummy package and definitions files in 0.8 dir
+        /// Create empty packages and definitions dir in 0.9 folder
+        /// </summary>
+        /// <param name="userDataDir">return root folder</param>
+        private static void CreateMockDirectoriesAndFiles(out string userDataDir)
+        {
+            
+            string tempPath = Path.GetTempPath();
```

Comment:
Thanks for adding the tests. Do not use `Path.GetTempPath` here (it accumulates temporary directories on the system), use `this.TempFolder` which will be destroyed at the end of each test run.